rir/src/compiler/translations/pir_2_rir/stack_use.cpp

```diff
@@ -46,12 +49,16 @@ void StackUseAnalysisState::AbstractStack::matchContents(
         if (operator[](i) != other.operator[](i)) {
             if (operator[](i)->isInstruction() &&
                 other.operator[](i)->isInstruction()) {
-                auto my = Instruction::Cast(operator[](i))->hasSingleUse();
-                auto their =
-                    Instruction::Cast(other.operator[](i))->hasSingleUse();
-                assert(my && Phi::Cast(my));
-                assert(their && Phi::Cast(their));
-                assert(my == their && "Stack mismatch.");
+                assert(Instruction::Cast(operator[](i))->hasSingleUse() &&
```

Comment:
ehm, how about using a temporary variable to store `operator[](i)`? (which is already the imho the ugly version of `(*this)[i]`, but that's not from your pr, ik)